homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,484 @@
+                    item_id,
+                    entity_setting["device_class"],
+                    entity_setting["icon"],
+                    entity_setting["unit"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class OmnilogicSensor(OmniLogicEntity):
+    """Defines an Omnilogic sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        device_class: str,
+        icon: str,
+        unit: str,
+        item_id: tuple,
+        state_key: str,
+    ):
+        """Initialize Entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            item_id=item_id,
+            icon=icon,
+        )
+
+        backyard_id = item_id[:2]
+        unit_type = coordinator.data[backyard_id].get("Unit-of-Measurement")
+
+        self._state = None
+        self._unit_type = unit_type
+        self._device_class = device_class
+        self._unit = unit
+        self._state_key = state_key
+
+    @property
+    def device_class(self):
+        """Return the device class of the entity."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self):
+        """Return the right unit of measure."""
+        return self._unit
+
+    @property
+    def force_update(self):
+        """Force update."""
+        return True
+
+
+class OmniLogicTemperatureSensor(OmnilogicSensor):
+    """Define an OmniLogic Temperature (Air/Water) Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the temperature sensor."""
+        sensor_data = self.coordinator.data[self._item_id].get(self._state_key)
+
+        temp_return = int(sensor_data)
+        temp_state = int(sensor_data)
+        unit_of_measurement = TEMP_FAHRENHEIT
+        if self._unit_type == "Metric":
+            temp_return = round((temp_return - 32) * 5 / 9, 1)
+            unit_of_measurement = TEMP_CELSIUS
+
+        if int(sensor_data) == -1:
+            temp_return = None
+            temp_state = None
+
+        self._attrs["hayward_temperature"] = temp_return
+        self._attrs["hayward_unit_of_measure"] = unit_of_measurement
+        if temp_state is not None:
+            self._state = float(temp_state)
```

Comment:
Just store the state in a local variable `state`.

Reply:
I have addressed I believe.

Note we are intentionally setting the state to None. In the case where the pool pump is off, the Omnilogic system can not return an accurate temperature. In that case we should return unavailable since there are no temperature readings happening.

Reply:
Make sure to differentiate between unknown state (`None`) and unavailable state (`available` set to `False`).

Reply:
Sorry - correct. We are returning None for unknown.

Reply:
Unknown state means that we haven't received any info yet and don't know what the state will be. Unavailable state means we know that the device or service is offline, for some reason.

Here it sounds like we know that the device is offline. Is that correct?

Reply:
The device is online, but we aren't receiving valid temperature data because the pump isn't running so we don't know what the state is. I feel like by the definition above this should be unknown (None).

Reply:
Ok.